### ISSUE TITLE
api.connection:083 and Display does not work

### ISSUE THREAD

**Issue by minimag450** [esphome/issues#3151]
### The problem

I can't find the problem. I have a very simple skech, but the temperature from the sensor is not shown on the display. I get the error [api.connection:083] ... Connection closed. What I also don't understand are the log files. If I display the log file when the ESP is plugged into the Raspberry, nothing else comes up. The time is also not displayed, only the elapsed time, for example 1:05:30. If I then create the log file via wireless, the clock works after a few seconds (I think when the WLAN connection is established) but the temperature is still not displayed. I have already tried the other answers in the issues, but it still doesn't work. I have also tried a D1 mini and an ESP8266 with the correct PINs, but the temperature is not displayed.

Translated with www.DeepL.com/Translator (free version)

### Which version of ESPHome has the issue?

2022.3.0

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.3.5

### What platform are you using?

ESP32

### Board

ESP32-WROOM-32

### Example YAML snippet

```yaml
esphome:
  name: anzeige1zoll
  
esp32:
  board: nodemcu-32s
  
wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_pass
  

#    ssid: "anzeige1zoll Fallback Hotspot"
#    password: "secretpassword"
    
captive_portal:

logger:
   
api:
   password: !secret password 

ota:
   password: !secret password
  
time:
  - platform: homeassistant
    id: esptime

sensor:
  - platform: homeassistant
    id: aussen_temperatur
    entity_id: sensor.temperaturmelder_funk_2
#    internal: true

color:
  - id: my_red
    red: 90%
    green: 80%
    blue: 80%

font:
  - file: 'arial.ttf'
    id: font1
    size: 32

spi: 
   clk_pin: 18
   mosi_pin: 23
   miso_pin: 19

display:
  - platform: ili9341
    model: TFT 2.4
    cs_pin: 14
    dc_pin: 27
    led_pin: 32
    reset_pin: 33
    rotation: 270
    lambda: |-
     it.fill(COLOR_OFF);
      it.strftime(165, 5, id(font1), id(my_red), TextAlign::TOP_CENTER, "%A %d.%m.%y", id(esptime).now());
      it.strftime(45, 55, id(font1), id(my_red), TextAlign::TOP_LEFT, "%H:%M:%S", id(esptime).now());
      
      it.print( 3, 120, id(font1), id(my_red), TextAlign::TOP_LEFT, "Aussen");
      if (id(aussen_temperatur).has_state()) {
       it.printf(318, 120, id(font1), id(my_red), TextAlign::TOP_RIGHT, "%.1f, °C", id(aussen_temperatur).state);
      }
```

### Anything in the logs that might be useful for us?

```txt
INFO Reading configuration /config/esphome/anzeige1zoll.yaml...
INFO Detected timezone 'Europe/Berlin'
INFO Generating C++ source...
INFO Compiling app...
Processing anzeige1zoll (board: nodemcu-32s; framework: arduino; platform: platformio/espressif32 @ 3.5.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- <AsyncTCP-esphome> 1.2.2
|-- <WiFi> 1.0
|-- <FS> 1.0
|-- <Update> 1.0
|-- <ESPAsyncWebServer-esphome> 2.1.0
|   |-- <AsyncTCP-esphome> 1.2.2
|-- <DNSServer> 1.1.0
|-- <ESPmDNS> 1.0
|-- <SPI> 1.0
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/api_connection.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/api_frame_helper.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/api_pb2.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/api_pb2_service.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/api_server.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/list_entities.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/proto.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/subscribe_state.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/user_services.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/api/util.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/display/display_buffer.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/esp32/core.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/esp32/gpio_arduino.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/esp32/gpio_idf.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/esp32/preferences.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/homeassistant/sensor/homeassistant_sensor.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/homeassistant/time/homeassistant_time.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/ili9341/ili9341_display.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/logger/logger.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/md5/md5.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/mdns/mdns_component.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/mdns/mdns_esp32_arduino.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/mdns/mdns_esp8266.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/mdns/mdns_esp_idf.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/network/util.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/ota/ota_backend_arduino_esp32.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/ota/ota_backend_arduino_esp8266.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/ota/ota_backend_esp_idf.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/ota/ota_component.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/sensor/automation.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/sensor/filter.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/sensor/sensor.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/socket/bsd_sockets_impl.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/socket/lwip_raw_tcp_impl.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/socket/socket.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/spi/spi.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/time/automation.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/time/real_time_clock.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/wifi/wifi_component.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/wifi/wifi_component_esp32_arduino.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/wifi/wifi_component_esp8266.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/components/wifi/wifi_component_esp_idf.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/application.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/color.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/component.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/controller.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/entity_base.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/helpers.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/log.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/scheduler.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/esphome/core/util.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/src/main.cpp.o
Generating partitions /data/anzeige1zoll/.pioenvs/anzeige1zoll/partitions.bin
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib528/AsyncTCP-esphome/AsyncTCP.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/ETH.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFi.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiAP.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib528/libAsyncTCP-esphome.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiClient.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiGeneric.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiMulti.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiSTA.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiScan.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiServer.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/WiFi/WiFiUdp.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib01c/FS/FS.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib01c/FS/vfs_api.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libbc6/Update/HttpsOTAUpdate.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib64d/libWiFi.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libbc6/Update/Updater.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/ESPAsyncWebServer-esphome/AsyncEventSource.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/ESPAsyncWebServer-esphome/AsyncWebSocket.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib01c/libFS.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/ESPAsyncWebServer-esphome/WebAuthentication.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/libbc6/libUpdate.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/ESPAsyncWebServer-esphome/WebHandlers.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/ESPAsyncWebServer-esphome/WebRequest.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/ESPAsyncWebServer-esphome/WebResponses.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/ESPAsyncWebServer-esphome/WebServer.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib333/DNSServer/DNSServer.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib333/libDNSServer.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib915/ESPmDNS/ESPmDNS.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib6b4/SPI/SPI.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/libFrameworkArduinoVariant.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/Esp.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/libc74/libESPAsyncWebServer-esphome.a
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib6b4/libSPI.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/FunctionalInterrupt.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/HardwareSerial.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/IPAddress.cpp.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/lib915/libESPmDNS.a
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/IPv6Address.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/MD5Builder.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/Print.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/Stream.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/StreamString.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/WMath.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/WString.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/base64.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/cbuf.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-adc.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-bt.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-cpu.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-dac.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-gpio.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-i2c.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-ledc.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-log.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-matrix.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-misc.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-psram.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-rmt.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-sigmadelta.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-spi.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-time.c.o
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-spi.c: In function 'spiTransferBytesNL':
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-spi.c:922:39: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
                 uint8_t * last_out8 = &result[c_longs-1];
                                       ^
/data/cache/platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-spi.c:923:40: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
                 uint8_t * last_data8 = &last_data;
                                        ^
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-timer.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-touch.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/esp32-hal-uart.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/libb64/cdecode.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/libb64/cencode.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/main.cpp.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/stdlib_noniso.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/wiring_pulse.c.o
Compiling /data/anzeige1zoll/.pioenvs/anzeige1zoll/FrameworkArduino/wiring_shift.c.o
Archiving /data/anzeige1zoll/.pioenvs/anzeige1zoll/libFrameworkArduino.a
Linking /data/anzeige1zoll/.pioenvs/anzeige1zoll/firmware.elf
RAM:   [=         ]  12.8% (used 41828 bytes from 327680 bytes)
Flash: [=====     ]  50.6% (used 927674 bytes from 1835008 bytes)
Building /data/anzeige1zoll/.pioenvs/anzeige1zoll/firmware.bin
esp32_create_combined_bin(["/data/anzeige1zoll/.pioenvs/anzeige1zoll/firmware.bin"], ["/data/anzeige1zoll/.pioenvs/anzeige1zoll/firmware.elf"])
Wrote 0xf2820 bytes to file /data/anzeige1zoll/.pioenvs/anzeige1zoll/firmware-factory.bin, ready to flash to offset 0x0
======================== [SUCCESS] Took 121.20 seconds ========================
INFO Successfully compiled program.
esptool.py v3.2
Serial port /dev/ttyUSB1
Connecting....
Chip is ESP32-D0WDQ6 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: f0:08:d1:d2:37:4c
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Flash will be erased from 0x00010000 to 0x000f2fff...
Flash will be erased from 0x00001000 to 0x00005fff...
Flash will be erased from 0x00008000 to 0x00008fff...
Flash will be erased from 0x0000e000 to 0x0000ffff...
Compressed 927776 bytes to 530742...
Wrote 927776 bytes (530742 compressed) at 0x00010000 in 13.2 seconds (effective 563.8 kbit/s)...
Hash of data verified.
Compressed 17104 bytes to 11191...
Wrote 17104 bytes (11191 compressed) at 0x00001000 in 0.6 seconds (effective 219.5 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 144...
Wrote 3072 bytes (144 compressed) at 0x00008000 in 0.1 seconds (effective 323.9 kbit/s)...
Hash of data verified.
Compressed 8192 bytes to 47...
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.1 seconds (effective 450.6 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
INFO Successfully uploaded program.
INFO Starting log output from /dev/ttyUSB1 with baud rate 115200
[20:54:49][I][logger:214]: Log initialized
[20:54:49][C][ota:461]: There have been 0 suspected unsuccessful boot attempts.
[20:54:49][D][esp32.preferences:114]: Saving preferences to flash...
[20:54:49][I][app:029]: Running through setup()...
[20:54:49][C][spi:023]: Setting up SPI bus...
[20:54:50][C][wifi:037]: Setting up WiFi...
[20:54:50][D][wifi:373]: Starting scan...
[20:54:51][D][wifi:388]: Found networks:
[20:54:51][I][wifi:432]: - 'FRITZ!Box 7490' [redacted]▂▄▆█
[20:54:51][D][wifi:433]:     Channel: 6
[20:54:51][D][wifi:434]:     RSSI: -70 dB
[20:54:51][I][wifi:432]: - 'FRITZ!Box 7490' [redacted]▂▄▆█
[20:54:51][D][wifi:433]:     Channel: 13
[20:54:51][D][wifi:434]:     RSSI: -77 dB
[20:54:51][D][wifi:437]: - [redacted] [redacted]▂▄▆█
[20:54:51][D][wifi:437]: - [redacted] [redacted]▂▄▆█
[20:54:51][I][wifi:248]: WiFi Connecting to 'FRITZ!Box 7490'...
[20:54:52][I][wifi:505]: WiFi Connected!
[20:54:52][C][wifi:353]:   Local MAC: F0:08:D1:D2:37:4C
[20:54:52][C][wifi:354]:   SSID: [redacted]
[20:54:52][C][wifi:355]:   IP Address: 192.168.178.58
[20:54:52][C][wifi:357]:   BSSID: [redacted]
[20:54:52][C][wifi:358]:   Hostname: 'anzeige1zoll'
[20:54:52][C][wifi:360]:   Signal strength: -68 dB ▂▄▆█
[20:54:52][C][wifi:364]:   Channel: 6
[20:54:52][C][wifi:365]:   Subnet: 255.255.255.0
[20:54:52][C][wifi:366]:   Gateway: 192.168.178.1
[20:54:52][C][wifi:367]:   DNS1: 192.168.178.1
[20:54:52][C][wifi:368]:   DNS2: 0.0.0.0
[20:54:52][C][ota:085]: Over-The-Air Updates:
[20:54:52][C][ota:086]:   Address: anzeige1zoll.local:3232
[20:54:52][C][ota:089]:   Using Password.
[20:54:52][C][api:025]: Setting up Home Assistant API server...
[20:54:52][I][app:062]: setup() finished successfully!
[20:54:52][I][app:102]: ESPHome version 2022.3.0 compiled on Mar 17 2022, 20:53:23
[20:54:52][C][wifi:491]: WiFi:
[20:54:52][C][wifi:353]:   Local MAC: F0:08:D1:D2:37:4C
[20:54:52][C][wifi:354]:   SSID: [redacted]
[20:54:52][C][wifi:355]:   IP Address: 192.168.178.58
[20:54:52][C][wifi:357]:   BSSID: [redacted]
[20:54:52][C][wifi:358]:   Hostname: 'anzeige1zoll'
[20:54:52][C][wifi:360]:   Signal strength: -68 dB ▂▄▆█
[20:54:52][C][wifi:364]:   Channel: 6
[20:54:52][C][wifi:365]:   Subnet: 255.255.255.0
[20:54:52][C][wifi:366]:   Gateway: 192.168.178.1
[20:54:52][C][wifi:367]:   DNS1: 192.168.178.1
[20:54:52][C][wifi:368]:   DNS2: 0.0.0.0
[20:54:52][C][logger:233]: Logger:
[20:54:52][C][logger:234]:   Level: DEBUG
[20:54:52][C][logger:235]:   Log Baud Rate: 115200
[20:54:52][C][logger:236]:   Hardware UART: UART0
[20:54:52][C][spi:097]: SPI bus:
[20:54:52][C][spi:098]:   CLK Pin: GPIO18
[20:54:52][C][spi:099]:   MISO Pin: GPIO19
[20:54:52][C][spi:100]:   MOSI Pin: GPIO23
[20:54:52][C][spi:102]:   Using HW SPI: YES
[20:54:53][C][homeassistant.time:010]: Home Assistant Time:
[20:54:53][C][homeassistant.time:011]:   Timezone: 'CET-1CEST,M3.5.0,M10.5.0/3'
[20:54:53][C][ili9341:030]: ili9341
[20:54:53][C][ili9341:030]:   Rotations: 270 °
[20:54:53][C][ili9341:030]:   Dimensions: 320px x 240px
[20:54:53][C][ili9341:031]:   Width: 240, Height: 320,  Rotation: 270
[20:54:53][C][ili9341:032]:   Reset Pin: GPIO33
[20:54:53][C][ili9341:033]:   DC Pin: GPIO27
[20:54:53][C][ili9341:035]:   Backlight Pin: GPIO32
[20:54:53][C][ili9341:036]:   Update Interval: 1.0s
[20:54:53][C][captive_portal:088]: Captive Portal:
[20:54:53][C][mdns:084]: mDNS:
[20:54:53][C][mdns:085]:   Hostname: anzeige1zoll
[20:54:53][C][ota:085]: Over-The-Air Updates:
[20:54:53][C][ota:086]:   Address: anzeige1zoll.local:3232
[20:54:53][C][ota:089]:   Using Password.
[20:54:53][C][api:138]: API Server:
[20:54:53][C][api:139]:   Address: anzeige1zoll.local:6053
[20:54:53][C][api:143]:   Using noise encryption: NO
[20:54:53][C][homeassistant.sensor:030]: Homeassistant Sensor 'aussen_temperatur'
[20:54:53][C][homeassistant.sensor:030]:   State Class: ''
[20:54:53][C][homeassistant.sensor:030]:   Unit of Measurement: ''
[20:54:53][C][homeassistant.sensor:030]:   Accuracy Decimals: 1
[20:54:53][C][homeassistant.sensor:031]:   Entity ID: 'sensor.temperaturmelder_funk_2'
[20:54:55][D][api:102]: Accepted ::FFFF:C0A8:B251
[20:54:55][W][api.connection:083]: Home Assistant 2022.3.5 (::FFFF:C0A8:B251): Connection closed
```

### Additional information

Logfile when connection is WLAN
```
INFO Reading configuration /config/esphome/anzeige1zoll.yaml...
INFO Detected timezone 'Europe/Berlin'
INFO Starting log output from anzeige1zoll.local using esphome API
INFO Successfully connected to anzeige1zoll.local
[20:56:04][I][app:102]: ESPHome version 2022.3.0 compiled on Mar 17 2022, 20:53:23
[20:56:04][C][wifi:491]: WiFi:
[20:56:04][C][wifi:353]:   Local MAC: F0:08:D1:D2:37:4C
[20:56:04][C][wifi:354]:   SSID: [redacted]
[20:56:04][C][wifi:355]:   IP Address: 192.168.178.58
[20:56:04][C][wifi:357]:   BSSID: [redacted]
[20:56:04][C][wifi:358]:   Hostname: 'anzeige1zoll'
[20:56:04][C][wifi:360]:   Signal strength: -68 dB ▂▄▆█
[20:56:04][C][wifi:364]:   Channel: 6
[20:56:04][C][wifi:365]:   Subnet: 255.255.255.0
[20:56:04][C][wifi:366]:   Gateway: 192.168.178.1
[20:56:04][C][wifi:367]:   DNS1: 192.168.178.1
[20:56:04][C][wifi:368]:   DNS2: 0.0.0.0
[20:56:04][C][logger:233]: Logger:
[20:56:04][C][logger:234]:   Level: DEBUG
[20:56:04][C][logger:235]:   Log Baud Rate: 115200
[20:56:04][C][logger:236]:   Hardware UART: UART0
[20:56:05][C][spi:097]: SPI bus:
[20:56:05][C][spi:098]:   CLK Pin: GPIO18
[20:56:05][C][spi:099]:   MISO Pin: GPIO19
[20:56:05][C][spi:100]:   MOSI Pin: GPIO23
[20:56:05][C][spi:102]:   Using HW SPI: YES
[20:56:05][C][homeassistant.time:010]: Home Assistant Time:
[20:56:05][C][homeassistant.time:011]:   Timezone: 'CET-1CEST,M3.5.0,M10.5.0/3'
[20:56:05][C][ili9341:030]: ili9341
[20:56:05][C][ili9341:030]:   Rotations: 270 °
[20:56:05][C][ili9341:030]:   Dimensions: 320px x 240px
[20:56:05][C][ili9341:031]:   Width: 240, Height: 320,  Rotation: 270
[20:56:05][C][ili9341:032]:   Reset Pin: GPIO33
[20:56:05][C][ili9341:033]:   DC Pin: GPIO27
[20:56:05][C][ili9341:035]:   Backlight Pin: GPIO32
[20:56:05][C][ili9341:036]:   Update Interval: 1.0s
[20:56:05][C][captive_portal:088]: Captive Portal:
[20:56:05][C][mdns:084]: mDNS:
[20:56:05][C][mdns:085]:   Hostname: anzeige1zoll
[20:56:05][C][ota:085]: Over-The-Air Updates:
[20:56:05][C][ota:086]:   Address: anzeige1zoll.local:3232
[20:56:05][C][ota:089]:   Using Password.
[20:56:05][C][api:138]: API Server:
[20:56:05][C][api:139]:   Address: anzeige1zoll.local:6053
[20:56:05][C][api:143]:   Using noise encryption: NO
[20:56:05][C][homeassistant.sensor:030]: Homeassistant Sensor 'aussen_temperatur'
[20:56:05][C][homeassistant.sensor:030]:   State Class: ''
[20:56:05][C][homeassistant.sensor:030]:   Unit of Measurement: ''
[20:56:05][C][homeassistant.sensor:030]:   Accuracy Decimals: 1
[20:56:05][C][homeassistant.sensor:031]:   Entity ID: 'sensor.temperaturmelder_funk_2'
[20:56:56][D][api:102]: Accepted ::FFFF:C0A8:B251
[20:56:56][W][api.connection:083]: Home Assistant 2022.3.5 (::FFFF:C0A8:B251): Connection closed
```

**Comment by ThijsVredenbregt** [comment]
I also have this issue

**Comment by davidsat** [comment]
I also have this issue. Started around 3-4 weeks ago. I see "connection closed" messages every minute, and ESPHOME does send the device info to Home Assistant.

**Comment by HumanSkunk** [comment]
Same issue here. In the end as a test I flashed the same board with a default config with nothing attached. I can see logs when connected locally, but they always end with "Connection closed". The board I am having issues with has been refreshed a few times with slightly different yaml configs and a different device name.

The board I am having issues with never originally showed up in HA as a device (which every other esp configured through ESPHome has.)

I am going to try a completely fresh board which has never been configured nor connected  tomorrow and see what happens.

**Comment by ThijsVredenbregt** [comment]
this started working auto-magically yesterday. Dont know what changed...

**Comment by rafalyesware** [comment]
I had this issue as a result of renaming a `esphome-tester` node (which was initially just that) to `basement-minisplit` in the ESPHome dashboard.  I'm not sure how, but the rename created the new `basement-minisplit` entity and still left the old `esphome-tester` one.  It's possible that I then tried to rename the device as well in Settings > Devices, but that part I'm less sure of.

Unfortunately to fix it, I had to hand-edit a bunch of things in `/config/.storage` (device registry, entity registry and/or entities as well as the `esphome.<device-id>` file) to get rid of the duplicate and finish the rename.  Ugh, but at least it's all happy now.
